cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java

```diff
@@ -560,6 +580,55 @@ private String pg(CQLTermNode node) throws QueryValidationException {
     return index2sql(node.getIndex(), node);
   }
 
+  private String pgSubQuery(CQLTermNode node, List<DbFkInfo> fks, boolean childToParent) throws QueryValidationException {
+    String currentTableName = dbTable.getTableName();
+    Table targetTable = null;
+
+    StringBuilder sb = new StringBuilder();
+    if (childToParent) {
+      // child to parent
+      targetTable = DbSchemaUtils.getTable(dbSchema, fks.get(fks.size() -1).getTargetTable());
+      for (DbFkInfo fk : fks) {
+        sb.append("(" + index2sqlText(currentTableName + ".jsonb", fk.getField()) + ")::UUID IN ");
+        sb.append(" ( SELECT " + PK_COLUMN_NAME + " from " + fk.getTargetTable() + " WHERE ");
+        currentTableName = fk.getTargetTable();
+      }
+    } else {
+      // parent to child
+      targetTable = DbSchemaUtils.getTable(dbSchema, fks.get(0).getTable());
+      for (int i = fks.size() - 1 ; i >= 0; i--) {
+        DbFkInfo fk = fks.get(i);
+        sb.append(currentTableName + "." + PK_COLUMN_NAME + " IN ");
+        sb.append(" ( SELECT " + "(" + index2sqlText(fk.getTable() + ".jsonb", fks.get(i).getField()) + ")::UUID");
+        sb.append(" from " + fk.getTable() + " WHERE ");
+        currentTableName = fk.getTable();
+      }
+    }
+    String [] foreignTarget = node.getIndex().split("\\.", 2);
+    sb.append(indexNodeForForeignTable(node, targetTable, foreignTarget));
+    for (int i = 0; i < fks.size(); i++) {
+      sb.append(")");
+    }
+    return sb.toString();
+  }
+
+  private String indexNodeForForeignTable(CQLTermNode node, Table targetTable, String[] foreignTarget)
+      throws QueryValidationException {
```

Comment:
Can you add a test to ForeignKeyGenerationIT that ensures that this also works for nested fields?
For example
* 1 nesting level like in user: personal.lastName https://s3.amazonaws.com/foliodocs/api/mod-users/users.html#users_post
* 2 nesting levels like in organisations: edi.ediFtp.username https://s3.amazonaws.com/foliodocs/api/mod-organizations-storage/organization.html#organizations_storage_organizations_post
Do not create user and organisations schemas, create such fields in one of our existing unit test schemas.